File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {} # declaring hashmap
        for i in range(len(nums)): # for loop to iterate through nums
            # Storing the complement/remaining number to reach target
            complement = target - nums[i]
            if complement in hashmap: # If complement in hashmap
                # return the complement and the current iteration num from nums
                return [i, hashmap[complement]] 
            # If not, 
            # add the current iteration of hashmap[i] = current iteration value of i
            hashmap[nums[i]] = i 

        return [] # If no matches are found return empty array

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> digits = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            digits.put(nums[i], i);
        }

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];

            if (digits.containsKey(complement) && digits.get(complement) != i) {
                return new int[]{i, digits.get(complement)};
            }
        }
        return null;
    }  
}